From ed85b855e1bcdb814b3dca7f6c768b86e6353fe7 Mon Sep 17 00:00:00 2001
From: "Lan Yixun (dlan)" <dennis.yxun@gmail.com>
Date: Thu, 25 Jul 2013 10:04:23 +0800
Subject: [PATCH] fix occasional build error when enable parallel compiling

the problem here we pass "-jX" ( X> 1) to "make", and "rpc_subs.c"
occasionally fail to compile due to it can't find "mount.h",
the root cause it that "mount.h" haven't generated by that time,
add this patch should fix this problem.
to make rpc_subs.o explicitly depend on "mount.h".

---
error message:

x86_64-pc-linux-gnu-gcc -O2 -pipe -march=core2  -D_REENTRANT
-D_FILE_OFFSET_BITS=64 -I/usr/include/tirp  c -D_REENTRANT
-D_FILE_OFFSET_BITS=64 -I/usr/include/tirpc -I../include -fPIC
-D_GNU_SOURCE -DAUTOFS_MA  P_DIR=\"/etc/autofs\"
-DAUTOFS_CONF_DIR=\"/etc/conf.d\" -c rpc_subs.c
rpc_subs.c:46:19: fatal error: mount.h: No such file or directory
  #include "mount.h"
                   ^
compilation terminated.
make[1]: *** [rpc_subs.o] Error 1
make[1]: *** Waiting for unfinished
jobs....

Signed-off-by: Lan Yixun (dlan) <dennis.yxun@gmail.com>
---
 lib/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Makefile b/lib/Makefile
index 5418009..314779b 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -70,6 +70,8 @@ nss_tok.o: nss_tok.c nss_parse.tab.h
 
 nss_parse.tab.o: nss_parse.tab.c nss_parse.tab.h
 
+rpc_subs.o: mount.h
+
 install: all
 
 clean:
-- 
1.8.3.2

